File: UniversityApi.API/DataBase/Configuration/ProfessorConfiguration.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System.Reflection.Emit;
using UniversityApi.API.DataBase.Entities;

namespace UniversityApi.API.DataBase.Configuration
{
    public class ProfessorConfiguration : IEntityTypeConfiguration<EntityProfessor>
    {
        public void Configure(EntityTypeBuilder<EntityProfessor> builder)
        {
            builder.HasData(
                new EntityProfessor
                {
                    EntityPersonID = Guid.Parse("18b517f2-b789-4230-e63c-08dbbf4015ce"),
                    FirstName = "Danuta",
                    LastName = "Dobrzycka",
                    PESEL = "14444444444",
                    Birthday = new DateTime(2000, 1, 1),
                    Motherland = "Poland",
                    IdCode = "11111",
                    Subject = "Programming",
                    FirstDayAtJob = new DateTime(2019, 10, 1),
                    Salary = 10000
                },
                new EntityProfessor
                {
                    EntityPersonID = Guid.Parse("28b517f2-b789-4230-e63c-08dbbf4015ce"),
                    FirstName = "Eustachy",
                    LastName = "Epoletnik",
                    PESEL = "15555555555",
                    Birthday = new DateTime(1999, 2, 2),
                    Motherland = "Poland",
                    IdCode = "22222",
                    Subject = "Science",
                    FirstDayAtJob = new DateTime(2018, 10, 1),
                    Salary = 9000
                });
        }
    }
}

File: FsUniversityApi/Database/Configurations/ProfessorConfiguration.fs
namespace FsUniversityApi.Database.Configurations

open FsUniversityApi.Database.Entities.Professor
open Microsoft.EntityFrameworkCore
open Microsoft.EntityFrameworkCore.Metadata.Builders
open System
open FsUniversityApi.Database.Configurations

type ProfessorConfiguration() =
    let initialPersonInfo = PersonConfiguration.initialPersonInfos
    interface IEntityTypeConfiguration<Professor> with
        member this.Configure(builder : EntityTypeBuilder<Professor>) =
            builder.HasData(
                { PersonInfo = initialPersonInfo.Item(0); PersonInfoId = initialPersonInfo.Item(0).PersonInfoId; IdCode = "11111"; Subject = "Programming"; Salary = 5000; FirstDayAtJob = DateTime(2018, 10, 1) },
                { PersonInfo = initialPersonInfo.Item(1); PersonInfoId = initialPersonInfo.Item(1).PersonInfoId; IdCode = "22222"; Subject = "Mathematics"; Salary = 6000; FirstDayAtJob = DateTime(2020, 10, 5)  },
                { PersonInfo = initialPersonInfo.Item(2); PersonInfoId = initialPersonInfo.Item(2).PersonInfoId; IdCode = "33333"; Subject = "Physics"; Salary = 4500; FirstDayAtJob = DateTime(2005, 3, 10)  }
                ) |> ignore